dev-docs/RFCs/v8.1/gpu-accelerated-polygon-clipping.md

```diff
@@ -0,0 +1,153 @@
+
+`opts` object is same as `opts` argument of constructor. Once object is constructed, polygons can be updated.
+
+##### run(opts)
+
+* `opts.positionBuffer` (Buffer) : `Buffer` object containing position data of input points, must contain X and Y values.
+* `opts.count` (Integer) : Count of number of points to be processed.
+* `opts.resultBuffer` (Buffer): `Buffer` object where results will be stored. It should hold at least 3 elements (vec3) for each input point. Once this method is completed this buffer will contain a vec3 [`flag`, `id`, `index`]. Each element (vec3) of this buffer contains the result of corresponding point (vec2) from the `positionBuffer`.
+ - `flag` : 0 if the point is outside the polygon, 1 if inside any polygon.
+ - `id` : if the `flag` is `1`, this corresponds to the `id` of the polygon in which the point is inside.
+ - `index` : this is the index of the point, which is same of the element id. Its usage is discussed in `Future Work` section.
+
+##### delete()
+
+* Deletes all owned resources.
+
+Example :
+```js
+// Initialize
+
+const polygonFilter = new GPUPolygonFilter(gl);
+
+// update with polygons
+
+polygonFilter.update([
+  {
+    id: 5,
+    polygon: [[0, 0], [0, 1], [1, 1], [1, 0]]
+  }
+]);
+
+const count = 4;
+const positionBuffer = new Buffer(gl, new Float32Array([0.5, 0.5,  0, 0.5, 0, 3,  0.25, 0.75]));
+
+// holds 4 vec3 elements
+const resultBuffer = new Buffer(gl, new Float32Array(4 * 4 * 3));
+
+polygonFilter.run({positionBuffer, count, resultBuffer});
+
+// resultBuffer.getData() will return following array
+// [
+//   // flag, id, index
+//   1, 5, 0, // inside
+//   1, 5, 1, // inside
+//   0, 0, 2, // outside
+//   1, 5, 3 // inside
+// ]
+
+```
+
+## WebGL1 Support
+
+Praposed above new class `GPUPolygonFilter` is WebGL2 only. For WebGL1 applications, above shader module `textureFilter` can be used to and PIP test can be performed with in the application vertex shader. Based on the result vertices can be either discarded (in fragment shader) or any of the attribute (such as color) can be changed as per required visual effect.
+
+
+## Precision
+
+Compared to CPU, GPU's have low floating point precision and depending on the system, texture size is always limited to fixed number (usually 4K), hence there could be cases where a point very close the the boundary could have different result when PIP test is performed on CPU and GPU. To improve GPU's precession, `fp64-arithmetic` shader module can be used when generating texture coordinates in the `texturefilter` shader module.
+
+
+## Future Work
+
+Result of the PIP test returned by `GPUPolygonFilter` class resides in a `WebGLBuffer` object. Reading this `Buffer` object involves CPU and GPU sync. For a GPU bound application this causes a bottleneck, as it requires CPU and GPU sync. To avoid this, another `Transform Feedback` can be run on this Buffer object to produce the index buffer of point that have passed PIP test.
+
+
+## Performance
```

Comment:
I'd leave out all the text. The table speaks for itself.

Reply:
Removed some redundant text.